pac4j-oidc/src/main/java/org/pac4j/oidc/client/KeycloakOidcClient.java

```diff
@@ -0,0 +1,44 @@
+package org.pac4j.oidc.client;
+
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.util.CommonHelper;
+import org.pac4j.oidc.config.OidcConfiguration;
+import org.pac4j.oidc.profile.OidcProfileDefinition;
+import org.pac4j.oidc.profile.creator.OidcProfileCreator;
+import org.pac4j.oidc.profile.keycloak.KeycloakOidcProfile;
+
+/**
+ * <p>This class is the OpenID Connect client to authenticate users in Keycloak.</p>
+ * <p>This client needs two customParams in its configuration: </p>
+ * <ul>
+ *   <li>authUrl : Base auth server url, e.g., https://keycloak.example.com/auth </li>
+ *   <li>realm : keycloak realm</li>
+ * </ul>
+ *
+ * @author Julio Arrebola
+ * @since 2.0.0
+ */
+public class KeycloakOidcClient extends OidcClient<KeycloakOidcProfile> {
+
+    public KeycloakOidcClient() {
+    }
+
+    public KeycloakOidcClient(final OidcConfiguration configuration) {
+        super(configuration);
+    }
+
+    @Override
+    protected void internalInit(final WebContext context) {
+        CommonHelper.assertNotNull("configuration", getConfiguration());
+        String baseUri = getConfiguration().getCustomParam("authUrl");
```

Comment:
I've made a change moving those properties to the configuration class instead of the client to see how it looks. If you think this is better than use the client to set those configuration parameters I can do the same changes to the GenericOauth20Client; otherwise I'll move then to the client.